backends/apple/metal/metal_backend.py

```diff
@@ -108,34 +108,62 @@ def preprocess(
         options: dict[str, typing.Any] = {
```

Comment:
Is there a reason why we need metal_backend.py? It feels very much like cuda_backend.py

Reply:
Yes, I agree. Next week we should begin refactoring, and moving most of this code to a new aoti_backend.py. Same with most of the code in metal_backend.cpp/cuda_backend.cpp